File: internal/agent/setup/prompt.go
package setup

import (
	"fmt"
	"io/fs"
	"strings"
)

// Prompts loads markdown prompt files from an fs.FS. Each agent embeds its own
// prompts/ directory (//go:embed prompts/*.md) and passes the embed.FS here, so
// prompts stay as reviewable markdown next to the agent that uses them.
type Prompts struct {
	fsys fs.FS
}

// NewPrompts wraps a filesystem (typically an embed.FS rooted at the agent package).
func NewPrompts(fsys fs.FS) Prompts {
	return Prompts{fsys: fsys}
}

// Get returns the trimmed contents of prompts/<name>.md.
func (p Prompts) Get(name string) (string, error) {
	b, err := fs.ReadFile(p.fsys, "prompts/"+name+".md")
	if err != nil {
		return "", fmt.Errorf("read prompt %q: %w", name, err)
	}
	return strings.TrimSpace(string(b)), nil
}

// MustGet is Get but panics on error. Use at agent construction time, where a
// missing prompt is a programming error that should fail fast at startup.
func (p Prompts) MustGet(name string) string {
	s, err := p.Get(name)
	if err != nil {
		panic(err)
	}
	return s
}

File: python/automation_agent/agent/setup/prompt.py
"""Markdown prompt loader.

Each agent ships its own ``prompts/`` directory of markdown files and reads them
through :class:`Prompts`, so prompts stay reviewable next to the agent that uses
them. Prompts are loaded with ``importlib.resources`` against the agent's package.
"""

from __future__ import annotations

from importlib import resources


class Prompts:
    """Loads ``prompts/<name>.md`` from a given package anchor."""

    def __init__(self, anchor: str) -> None:
        """anchor is the dotted package name whose ``prompts/`` dir holds the files,
        e.g. ``"automation_agent.agent.summary"``."""
        self._anchor = anchor

    def get(self, name: str) -> str:
        """Return the trimmed contents of ``prompts/<name>.md``.

        Raises:
            FileNotFoundError / OSError: if the prompt file is missing.
        """
        try:
            path = resources.files(self._anchor).joinpath("prompts", f"{name}.md")
            return path.read_text(encoding="utf-8").strip()
        except (FileNotFoundError, ModuleNotFoundError) as exc:
            raise FileNotFoundError(f"read prompt {name!r}: {exc}") from exc

    def must_get(self, name: str) -> str:
        """Like :meth:`get`, but intended for agent-construction time where a missing
        prompt is a programming error that should fail fast at startup."""
        return self.get(name)
